arduino_examples/google_home_thermostat.ino

```diff
@@ -0,0 +1,196 @@
+#include <ArduinoJson.h>      // https://github.com/kakopappa/sinric/wiki/How-to-add-dependency-libraries
+#include <StreamString.h>
+#include <IRremoteESP8266.h>
+#include <ir_Samsung.h>
+
+ESP8266WiFiMulti WiFiMulti;
+WebSocketsClient webSocket;
+WiFiClient client;
+
+uint64_t heartbeatTimestamp = 0;
+bool isConnected = false;
+
+#define MyApiKey ""       // ApiKey Sinric
+#define MySSID ""         // TODO: Change to your Wifi network SSID
+#define MyWifiPassword "" // TODO: Change to your Wifi network password
+#define HEARTBEAT_INTERVAL 300000
+#define MyThermostatId "" // Your Thermostat ID
+
+//---------------Thermostat-------------------------
+
+IRSamsungAc ac(D2);
+
+//--------------------------------------------------
+
+void setTemp(String deviceId, int temp)
+{
+    if (deviceId == MyThermostatId) // Device ID of first device
+    {
+        Serial.print("Turn temp Thermostat id: ");
+        Serial.println(deviceId);
+        ac.setTemp(temp);
+        ac.send();
+    }
+}
+
+void turnOn(String deviceId)
+{
+    if (deviceId == MyThermostatId) // Device ID of first device
+    {
+        Serial.print("Turn on AR id: ");
+        Serial.println(deviceId);
+        ac.on();
+        ac.send();
+    }
+}
+void turnOff(String deviceId)
+{
+    if (deviceId == MyThermostatId) // Device ID of first device
+    {
+        Serial.print("Turn off AR id: ");
+        Serial.println(deviceId);
+        ac.off();
+        ac.send();
+    }
+}
+
+void webSocketEvent(WStype_t type, uint8_t *payload, size_t length)
+{
+    switch (type)
+    {
+    case WStype_DISCONNECTED:
+        isConnected = false;
+        Serial.printf("[WSc] Webservice disconnected from sinric.com!\n");
+        break;
+    case WStype_CONNECTED:
+    {
+        isConnected = true;
+        Serial.printf("[WSc] Service connected to sinric.com at url: %s\n", payload);
+        Serial.printf("Waiting for commands from sinric.com ...\n");
+    }
+    break;
+    case WStype_TEXT:
+    {
+        Serial.printf("[WSc] get text: %s\n", payload);
+
+#if ARDUINOJSON_VERSION_MAJOR == 5
+        DynamicJsonBuffer jsonBuffer;
+        JsonObject &json = jsonBuffer.parseObject((char *)payload);
+#endif
+#if ARDUINOJSON_VERSION_MAJOR == 6
+        DynamicJsonDocument json(1024);
+        deserializeJson(json, (char *)payload);
+#endif
+        String deviceId = json["deviceId"];
+        String action = json["action"];
+
+        if (action == "action.devices.commands.OnOff")
+        { // Switch
+            String value = json["value"]["on"];
+            Serial.println(value);
+
+            if (value == "true")
+            {
+                turnOn(deviceId);
+            }
+            else
+            {
+                turnOff(deviceId);
+            }
+        }
+        else if (action == "action.devices.commands.ThermostatTemperatureSetpoint")
+        {
+            float value = json["value"]["acTemperatureSetpoint"];
```

Comment:
Is this correct?

Reply:
Yes, when there is a change in the temperature of the thermostats, I receive the command "action.devices.commands.ThermostatTemperatureSetpoint", which gives me a float corresponding to the desired temperature!

Reply:
I mean float value = json["value"]["acTemperatureSetpoint"]; <--- is this correct ?

Reply:
Ah ok. Yes, that's correct!

Reply:
I recompiled and tested it here, it worked correctly!

Reply:
Thanks